src/utils/ProtectedRoute.jsx

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from 'react';
+import { Navigate } from 'react-router-dom';
+import { useAuth } from '../contexts/AuthContext';
+
+// eslint-disable-next-line react/prop-types
+const ProtectedRoute = ({ Component }) => {
+    const { currentUser } = useAuth();
+    const [isLoading, setIsLoading] = useState(true);
+    const [isAuthenticated, setIsAuthenticated] = useState(false);
+
+    useEffect(() => {
+        setIsAuthenticated(currentUser !== null);
+        setIsLoading(false);
+    }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'currentUser'. Either include it or remove the dependency array.